scripts/create-mcc-country-mappings.sh

```diff
@@ -14,75 +19,43 @@ For example
   123 45 AA Kingdom of Anonymous Aardwarks
   123 46 AA Kingdom of Anonymous Aardwarks
   456 78 RB Republic of Banana
-
-Options:
-  -h         Print this message"
+EOF
 }
 
-if [ $# != 1 ]; then
-    echo "${0##*/}: wrong number of arguments"
-    usage
-    exit 1
+if [ -z $1 ]; then
+  usage
+  exit 1
 fi
 
-if [ "$1" == "-h" ]; then
-    usage
-    exit
+if [ "$1" == "-h" -o "$1" == "--help" ] ; then
+  usage
+  exit 0
 fi
 
-if [ ! -f $1 ]; then
-    echo "${0##*/}: the file $1 does not exist"
-    usage
-    exit 1
+if [ ! -d $1 ] ; then
+  echo "ERROR: $1 is not a directory" >&2
+  exit 1
 fi
 
-TMPFILE1=`mktemp`
-TMPFILE2=`mktemp`
-
-# Fetch all lines that begin with either | or |===,
-# remove MCC codes 001 (test network) & 901 (international networks)
-grep -e "^|\|^===" $1 | grep -v "^|-\||}" \
-    | grep -v "^|[[:space:]]*001\|^|[[:space:]]*901" > $TMPFILE1
-
-# Remove wiki table controls, |, ||, [[, ]], ===, Replace " - " with ,
-cat $TMPFILE1 | sed 's/^|[[:space:]]*//' | sed 's/[[:space:]]*||[[:space:]]*/,/g' \
-    | sed 's/^=*[[:space:]]*\[\[//' | sed 's/]*[[:space:]]*-[[:space:]]*/,/' \
-    | sed 's/[[:space:]]=*$//' | sed 's/\[\[//g' | sed 's/\]\]//g' > $TMPFILE2
-
-# Replace &amp with &
-cat $TMPFILE2 | sed 's/&amp;/&/' > $TMPFILE1
-
-# Remove everyting starting with a &lt; (links)
-cat $TMPFILE1 | sed 's/&lt;.*//' > $TMPFILE2
-
-# Transform input
-#
-#   Kingdom of Anonymous Aardwarks,AA
-#   123,45,Operator A,..
-#   123,46,Operator B,..
-#   Republic of Banana,RB
-#   543,??,Not operational,...
-#   456,78,Operator C,....
-#
-# where lines have either a name of a country and ISO3166 country code,
-# or MCC, MNC, operator name, and stuff to
-#
-#  123 45 AA Kingdom of Anonymous Aardwarks
-#  123 46 AA Kingdom of Anonymous Aardwarks
-#  456 78 RB Republic of Banana
-#
-grep -v "?" $TMPFILE2 | awk '\
-BEGIN { \
-   country=""; iso="" \
-} { \
-   n=split($0, a, ","); \
-   if (n==2) { \
-      country=a[1]; iso=a[2]; \
-  } else { \
-      if (length(a[1]) == 3 && length(a[2]) > 0) \
-         print a[1]" "a[2]" "iso" "country; \
-  } \
-} END {}' | sort | uniq
-
-rm -f $TMPFILE1 $TMPFILE2
-exit
+for infile in $1/* ; do
```

Comment:
Should we sort the file list to get more stable output?

Reply:
This script processes the data line by line, so sorting inside it, would be... clunky. The data is already (kind of) sorted, since the wikipages list everything in alphabetical order and the script doesn't change the order. If one wants a perfectly consistent sort, I would suggest to just pipe the script output through sort, e.g. 
`sort --unique --key=3 --key=1 --key=2`
I could add this to the usage and comments...

Reply:
I meant only the file list, not the content of the files.